### ISSUE TITLE
SCRAM supports

### ISSUE THREAD

**Issue by Neustradamus** [vukoye/xmpp_dart#7]
I have seen a missing parts:
- SCRAM-SHA-256-PLUS, I think SCRAM-SHA-256 is not complete?
- https://github.com/vukoye/xmpp_dart/blob/master/lib/src/features/sasl/SaslAuthenticationFeature.dart
- https://github.com/vukoye/xmpp_dart/blob/master/test/connection_test.dart#L15
- https://github.com/vukoye/xmpp_dart/blob/master/test/feature_list_test.dart#L22

Maybe you can rename these files "ScramSha...", it is not only SCRAM-SHA-1 or have separate files "ScramSha1..." and "ScramSha256...":
- https://github.com/vukoye/xmpp_dart/blob/master/lib/src/features/sasl/ScramSha1SaslHandler.dart
- https://github.com/vukoye/xmpp_dart/blob/master/lib/src/features/sasl/ScramSha1SaslHandler_test.dart

**Comment by vukoye** [comment]
Yes you are right library currently supports following mechanisms
SCRAM-SHA-1
SCRAM-SHA-256
PLAIN

**Comment by Neustradamus** [comment]
It is in progress :)

Do not forget:
- https://github.com/vukoye/xmpp_dart/blob/master/test/feature_list_test.dart
- https://github.com/vukoye/xmpp_dart/blob/master/test/connection_test.dart

Can you look for add other SCRAM too for have:
- SCRAM-SHA-1
- SCRAM-SHA-1-PLUS
- SCRAM-SHA-224
- SCRAM-SHA-224-PLUS
- SCRAM-SHA-256
- SCRAM-SHA-256-PLUS
- SCRAM-SHA-384
- SCRAM-SHA-384-PLUS
- SCRAM-SHA-512
- SCRAM-SHA-512-PLUS
-SHA-3 family too?

----------------------------------------------------------------------

SCRAM-SHA-1(-PLUS):
- RFC5802: Salted Challenge Response Authentication Mechanism (SCRAM) SASL and GSS-API Mechanisms: https://tools.ietf.org/html/rfc5802
- RFC6120: Extensible Messaging and Presence Protocol (XMPP): Core: https://tools.ietf.org/html/rfc6120

SCRAM-SHA-256(-PLUS):
- RFC7677: SCRAM-SHA-256 and SCRAM-SHA-256-PLUS Simple Authentication and Security Layer (SASL) Mechanisms: https://tools.ietf.org/html/rfc7677 - since 2015-11-02
- RFC8600: Using Extensible Messaging and Presence Protocol (XMPP) for Security Information Exchange: https://tools.ietf.org/html/rfc8600 - since 2019-06-21: https://mailarchive.ietf.org/arch/msg/ietf-announce/suJMmeMhuAOmGn_PJYgX5Vm8lNA

-PLUS variants:
- RFC5056: On the Use of Channel Bindings to Secure Channels: https://tools.ietf.org/html/rfc5056
- RFC5929: Channel Bindings for TLS: https://tools.ietf.org/html/rfc5929
- Channel-Binding Types: https://www.iana.org/assignments/channel-binding-types/channel-binding-types.xhtml

LDAP:
- RFC5803: Lightweight Directory Access Protocol (LDAP) Schema for Storing Salted: Challenge Response Authentication Mechanism (SCRAM) Secrets: https://tools.ietf.org/html/rfc5803

HTTP:
- RFC7804: Salted Challenge Response HTTP Authentication Mechanism: https://tools.ietf.org/html/rfc7804

IANA:
- Simple Authentication and Security Layer (SASL) Mechanisms: https://www.iana.org/assignments/sasl-mechanisms/sasl-mechanisms.xhtml

**Comment by Neustradamus** [comment]
@vukoye: It has been solved?

**Comment by vukoye** [comment]
No sorry I closed the wrong one. 
Trying to do some cleanup
